### DISCUSSION TITLE
MQTT does not start

### DISCUSSION THREAD

**Discussion by oxygi** [MiczFlor/RPi-Jukebox-RFID#2263] [Q&A] [answered]
I was in a good mood and wanted to finally install the MQTT. I followed [all the steps](https://github.com/MiczFlor/RPi-Jukebox-RFID/tree/master/components/smart-home-automation/MQTT-protocol) and started the service, but I always get an error:

```
pi@raspberrypiJukebox:~ $ sudo python3 /home/pi/RPi-Jukebox-RFID/scripts/daemon_mqtt_client.py
Traceback (most recent call last):
  File "/home/pi/RPi-Jukebox-RFID/scripts/daemon_mqtt_client.py", line 593, in <module>
    client = mqtt.Client(config.get("mqttClientId"))
  File "/usr/local/lib/python3.9/dist-packages/paho/mqtt/client.py", line 766, in __init__
    raise ValueError(
ValueError: Unsupported callback API version: version 2.0 added a callback_api_version, see migrations.md for details
pi@raspberrypiJukebox:~ $
```

I am still on version [v2.5.0] and do not want to change everything :-)
Thank you very much in advance!

**Comment by AlvinSchiller** [comment]
Seems like a new Version (2.0.0) of the plugin was released last week, that included breaking changes.
https://github.com/eclipse/paho.mqtt.python/blob/v2.0.0/docs/migrations.rst

Can you try and install the latest version 1.6.1 that was used before?
`sudo python3 -m pip install --force-reinstall paho-mqtt==1.6.1`

**Reply by AlvinSchiller** [reply, marked as answer]
I added a PR for a fix. Could you maybe try out the changes? I dont have any mqtt stuff.
https://github.com/MiczFlor/RPi-Jukebox-RFID/pull/2265

**Reply by oxygi** [reply]
Hi, I have copied your changes into my system and restarted the service - it works:

pi@raspberrypiJukebox:~ $ sudo systemctl status phoniebox-mqtt-client
● phoniebox-mqtt-client.service - Phoniebox MQTT Client Service
     Loaded: loaded (/etc/systemd/system/phoniebox-mqtt-client.service; enabled; vendor preset: enabled)
     Active: **active (running)** since Sat 2024-02-17 09:44:07 CET; 28s ago
   Main PID: 3401 (python3)
      Tasks: 3 (limit: 1595)
        CPU: 3.149s
     CGroup: /system.slice/phoniebox-mqtt-client.service
             └─3401 python3 /home/pi/RPi-Jukebox-RFID/scripts/daemon_mqtt_client.py

**Comment by jgonzalez-89** [comment]
Specify Callback API Version
If the Paho MQTT client library allows for it, you might be able to specify the callback API version explicitly when creating the MQTT client object, to remain compatible with the older version your script was designed for. This could look something like:

```python
client = mqtt.Client(config.get("mqttClientId"), callback_api_version='1.0')
```
However, this is a hypothetical solution, as the exact parameter name and value might differ. You'll need to consult the library's documentation for the correct way to do this.
